feat: Add AssociatedResources to ModuleTemplate spec (#1923)
kyma-project/lifecycle-manager: docs/contributor/resources/03-moduletemplate.md

The `mandatory` field indicates whether the module is installed in all runtime clusters without any interaction from the user.
Mandatory modules do not appear in the Kyma CR `.status` and `.spec.modules`, furthermore they have the same configuration across all runtime clusters.

### **.spec.associatedResources**

> The `associatedResources` field is a list of module related resource definitions that usually should be cleaned up when uninstalling a module. 

Reviewer: ```suggestion
The `associatedResources` field is a list of module-related custom resource definitions (CRDs) that should be cleaned up during module deletion. 
```
should be cleaned up but are not? like leftovers? or just should be in any case? Can we say deletion? In the end-user docs we always say "add" or "delete" a module. Or maybe in that context "uninstalling" is better?

Author: It's an information for SRE in terms of deletion being stuck to have an indicator for what resources to look for on clean-up. KLM is not doing anything with it.

> The purpose of listing GVKs here is purely informational and does not introduce functional changes to the module.

Reviewer: ```suggestion
The list is purely informational and does not introduce functional changes to the module.
```
